### ISSUE TITLE
cuda_native experimental markers have no documented graduation criteria

### ISSUE THREAD

**Issue by s9213712** [s9213712/minicnn#10]
## Problem

`src/minicnn/cuda_native/capabilities.py` contains:

```python
'experimental': True,
'training_stable': False,
'backward_stable': False,
```

And `summary_status: 'experimental'` — but there is no documented definition of what these flags transitioning to `True`/`False` requires.

## Current state

- Training is marked experimental with broad warnings scattered across `_cli_training.py`, `capabilities.py`, `unified/trainer.py`, and `__init__.py`
- The `[EXPERIMENTAL]` prefix appears in 3 CLI parser help strings
- No documented checklist or threshold for graduation

## Risk

- The experimental label may persist indefinitely as it provides no pressure to define a boundary.
- Contributors can't tell whether their new op/feature is experimental by nature or just missing a stability declaration.
- Users see no clear path from "experimental" to "supported".

## Requested fix

Add a `docs/cuda_native_contract.md` (or section in `docs/architecture.md`) that defines:

1. **Stable op surface** — which ops are fully tested and publicly committed (e.g. Conv2d, BatchNorm2d eval, LayerNorm eval)
2. **Experimental op surface** — which ops have prototype status and may change (e.g. AMP, training backward)
3. **Graduation checklist** — what must be true before `training_stable` or `backward_stable` flips to `True`:
   - Parity test vs torch reference within tolerance
   - Test coverage in `tests/`
   - Smoke check passes

This makes the experimental boundary machine-verifiable instead of a free-floating label.

## Priority

P2 — without graduation criteria, the `experimental` markers are informational noise rather than a useful contract.

**Comment by s9213712** [comment]
Fixed on `cuda-native-expansion-plan` in commit `2f036ec`.

What changed:
- Added [docs/cuda_native_contract.md](https://github.com/s9213712/minicnn/blob/cuda-native-expansion-plan/docs/cuda_native_contract.md) with:
  - stable / beta / experimental surface definitions
  - graduation checklist for `training_stable` / `backward_stable`
  - meaning of the current `experimental` markers
- Linked the contract from `docs/cuda_native.md` and `docs/cuda_native_productionization_plan.md`.
- This now complements the existing machine-readable `support_tiers` / `support_tier_assessment` data already exposed by the CLI and artifacts.

This closes the missing-graduation-criteria gap.
